### PR TITLE
Bt discovery fix

### PR DESCRIPTION
**Pull request by csiki2**
With this patch the launchBTDiscovery function runs on the main core to send the mqtt messages via wifi.

### CODE REVIEW

File: main/main.ino [hunk 1]
```diff
@@ -1319,6 +1319,10 @@ void loop() {
       PilighttoMQTT();
 #endif
 #ifdef ZgatewayBT
+#  ifdef ZmqttDiscovery
+      if (disc)
+        launchBTDiscovery();
+#  endif
```
Comment (1technophile) [inline main/main.ino:1325]: With the discovery here, we are going to launch it without any delay or wait time

**Comment by 1technophile** [comment]
Thanks! I'm beginning the tests

**Comment by 1technophile** [comment]
I get some frequent Core 1 dump with this branch:
```
N: Received json : {"id":"D9:9E:1C","name":"OOO","rssi":-87,"distance":18.07764}
T: jsonPublishing
T: Pub json :{"id":"D9:9E:1C","name":"OOO","rssi":-87,"distance":18.07764} into custom topic: home/home_presence/OpenMQTTGateway_ESP32_IR_RF
N: Found 5 devices, scan number 5 end deinit controller
T: macAdr A4:00:21
T: isDisc 1
T: isWhtL 0
T: isBlkL 0
T: sensorModel 16
T: macAdr 58:2D:34:40:01:4C
T: isDisc 1
T: isWhtL 0
T: isBlkL 0
T: sensorModel 7
T: BT Task running on core: 0
N: Scan begin
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed)
Core 1 register dump:
PC      : 0x401887e8  PS      : 0x00060034  A0      : 0x80081480  A1      : 0x3ffbe7b0
A2      : 0x3ffc420c  A3      : 0x00000000  A4      : 0x3ffc4e94  A5      : 0x00000000
A6      : 0x3ffc4e90  A7      : 0xffffffff  A8      : 0x80081402  A9      : 0x00000001
A10     : 0x3ffbef38  A11     : 0x00000002  A12     : 0x00000000  A13     : 0x00000000
A14     : 0x00000000  A15     : 0x00000000  SAR     : 0x0000001a  EXCCAUSE: 0x00000007  
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000
Core 1 was running in ISR context:
EPC1    : 0x40087551  EPC2    : 0x00000000  EPC3    : 0x00000000  EPC4    : 0x401887e8

Backtrace: 0x401887e8:0x3ffbe7b0 0x4008147d:0x3ffbe7d0 0x400850e5:0x3ffbe7f0 0x4008754e:0x3ffba190 0x40083503:0x3ffba1b0 0x4008e3f1:0x3ffba1d0

Rebooting...
ets Jun  8 2016 00:22:57

rst:0x3 (SW_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
```

image

I tested the development branch so as to check that we did not miss something on the previous commits and it's working flawlessly. So it seems coming from these modifications.

**Comment by csiki2** [comment]
> I tested the development branch so as to check that we did not miss something on the previous commits and it's working flawlessly. So it seems coming from these modifications.

Interesting, it works for me for days now.
Any special setting, another module etc.?

**Comment by csiki2** [comment]
https://github.com/espressif/arduino-esp32/issues/3634#issuecomment-573423900
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/fatal-errors.html#guru-meditation-errors

Cache disabled but cached memory region accessed

In some situations ESP-IDF will temporarily disable access to external SPI Flash and SPI RAM via caches. For example, this happens with spi_flash APIs are used to read/write/erase/mmap regions of SPI Flash. In these situations, tasks are suspended, and interrupt handlers not registered with ESP_INTR_FLAG_IRAM are disabled. Make sure that any interrupt handlers registered with this flag have all the code and data in IRAM/DRAM. Refer to the SPI flash API documentation for more details.

Should we set something for some functions?
I don't have this issue at all.

**Comment by csiki2** [comment]
You can also increase the version of "esp32_platform = espressif32@1.12.2" to 3.1.0.
Other than that, I am clueless on what is going on.
If I add IRAM_ATTR to the onResult method I get tons of link errors that doesn't seem to be resolvable to me.
Changing the loglevel and increasing the BTQueueSize to 8 or 16 also help probably, but won't resolved the rootcause itself.
Do you have any extra flash? What is using it?

**Comment by 1technophile** [comment]
I have a track, I think we are missing an IRAM_ATTR on RCSwitch. I will test it and we will see how it goes.

**Comment by 1technophile** [comment]
In all the case doesn't seem linked with this PR, so let's merge it

**Comment by csiki2** [comment]
Thank you!

**Pull request merged**